mqtt/src/main/scala/Mqtt.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib.mqtt
+
+import akka.util.ByteString
+import akka.stream.stage._
+import org.eclipse.paho.client.mqttv3.{ MqttMessage => PahoMqttMessage, _ }
+import scala.util._
+import scala.language.implicitConversions
+
+case class MqttSourceSettings(
+  connectionSettings: MqttConnectionSettings,
+  topics:             Map[String, Int]
+)
+
+case class MqttConnectionSettings(
+  broker:      String,
+  clientId:    String,
+  persistence: MqttClientPersistence
+)
+
+case class MqttMessage(topic: String, payload: ByteString)
+
+trait MqttConnectorLogic { this: GraphStageLogic =>
+
+  import MqttConnectorLogic._
+
+  def connectionSettings: MqttConnectionSettings
+  def onConnect: AsyncCallback[IMqttAsyncClient]
+  def onMessage: AsyncCallback[MqttMessage]
+  def onConnectionLost: AsyncCallback[Throwable]
```

Comment:
Maybe define the AsyncCallbacks in this trait instead, and have the concrete connector just implement the synchronous callbacks, so source would implement `onMessage(message: MqTTMessage)`?